### PR TITLE
fix(ttid): End and measure TTID regardless current active span

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix

## :scroll: Description

- predecessor pr https://github.com/getsentry/sentry-react-native/pull/4018

This PR fixes possible edge cases where automatic ttid and its measurements might be missing when active span changed during the measuring.

## :green_heart: How did you test it?
unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by lucas-zimerman — APPROVED** [review]
Added a small suggestion but overall it looks good, LGTM!

File: src/js/tracing/utils.ts [hunk 1]
```diff
@@ -114,6 +109,24 @@ export function setSpanDurationAsMeasurement(name: string, span: Span): void {
...
+ * Sets the duration of the span as a measurement.
+ * Uses `setMeasurement` function from @sentry/core.
+ */
+export function setSpanDurationAsMeasurementOnTransaction(
+  transaction: TransactionType,
+  name: string,
+  span: Span,
+): void {
+  const spanEnd = spanToJSON(span).timestamp;
+  const spanStart = spanToJSON(span).start_timestamp;
```
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply src/js/tracing/utils.ts:122]: Yes, thanks. This is better.
